aws-lc-rs/src/evp_pkey.rs

```diff
@@ -158,6 +159,79 @@ impl LcPtr<EVP_PKEY> {
         // https://github.com/aws/aws-lc/blob/61503f7fe72457e12d3446853a5452d175560c49/crypto/refcount_lock.c#L29
         LcPtr::new(unsafe { EVP_PKEY_CTX_new(*self.as_mut_unsafe(), null_mut()) })
     }
+
+    pub(crate) fn marshal_raw_private_key(&self) -> Result<Vec<u8>, Unspecified> {
+        let mut size = 0;
+        if 1 != unsafe { EVP_PKEY_get_raw_private_key(*self.as_const(), null_mut(), &mut size) } {
+            return Err(Unspecified);
+        }
+        let mut buffer = vec![0u8; size];
+        let buffer_size = self.marshal_raw_private_to_buffer(&mut buffer)?;
+        debug_assert_eq!(buffer_size, size);
+        Ok(buffer)
+    }
+
+    pub(crate) fn marshal_raw_private_to_buffer(
+        &self,
+        buffer: &mut [u8],
+    ) -> Result<usize, Unspecified> {
+        let mut key_len = buffer.len();
+        if 1 == unsafe {
+            EVP_PKEY_get_raw_private_key(*self.as_const(), buffer.as_mut_ptr(), &mut key_len)
+        } {
+            Ok(key_len)
+        } else {
+            Err(Unspecified)
+        }
+    }
+
+    // pub(crate) fn marshal_raw_public_key(&self) -> Result<Vec<u8>, Unspecified> {
+    //     let mut size = 0;
+    //     if 1 != unsafe { EVP_PKEY_get_raw_public_key(*self.as_const(), null_mut(), &mut size) } {
+    //         return Err(Unspecified);
+    //     }
+    //     let mut buffer = vec![0u8; size];
+    //     let buffer_size = self.marshal_raw_public_to_buffer(&mut buffer)?;
+    //     debug_assert_eq!(buffer_size, size);
+    //     Ok(buffer)
+    // }
```

Comment:
Accidentally left in or intentional?

Reply:
It was intentional as my current draft of ML-DSA logic is using it, but I'm not it will be needed.  I'll remove this until we know it's needed.